src/scheduler/job_info.c

```diff
@@ -4577,17 +4577,17 @@ update_accruetype(int pbs_sd, server_info *sinfo,
 char*
 getaoename(selspec *select)
 {
+	int i = 0;
 	resource_req *req;
 
 	if (select == NULL)
 		return NULL;
 
-	/* we only need to look at 1st chunk since either all request aoe
-	 * or none request aoe.
-	 */
-	req = find_resource_req(select->chunks[0]->req, getallres(RES_AOE));
-	if (req != NULL)
-		return string_dup(req->res_str);
+	for(;i < select->total_chunks; i++) {
+		req = find_resource_req(select->chunks[i]->req, getallres(RES_AOE));
+		if (req != NULL)
+			return string_dup(req->res_str);
+	}
```

Comment:
Please initialize i within the for loop rather than when it is defined. Pleas add a space between for and the open paren. Do you want to return a list of AOEs, or just the first one you encounter that is not NULL? What if multiple AOEs are present? Please note that qsub isn't the only way jobs could be submitted.

Reply:
> Please initialize i within the for loop rather than when it is defined.
Done

> Pleas add a space between for and the open paren.
Done

>  Do you want to return a list of AOEs, or just the first one you encounter that is not NULL? What if multiple AOEs are present?
Yes, I intend to return the first encouneterd NON-NULL AOE. As per this RFE we are not allowing multiple AOE in single job request.

>  Please note that qsub isn't the only way jobs could be submitted.
In server side there is  a job validation for AOE request, which doesn't allow multiple type of aoe to be requested in single job request. This happens during job submission, so I think irrespective of what way we submit the job, qsub or not, a job with multiple type AOE request will get rejected.